cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -279,126 +281,168 @@ public void run(PrintStream out) throws Exception {
+                        //sleep for over 5 seconds because Prometheus only polls
+                        //every 5 seconds. If spark finishes and goes away immediately after completing,
+                        //Prometheus will never be able to poll for the final set of metrics for the spark-submit
+                        //The default promtheus config map was changed from 2 minute scrape interval to 5 seconds for spark pods
+                        Thread.sleep(7000);
+                    } else {
+                        LOG.info("Prometheus metrics not enabled");
+                    }
+                }
+                catch (NoSuchElementException e) {
+                    LOG.info("spark.ui.prometheus.enabled is not set");
+                }
 
-                OutputMetadataWriter writer = getOutputMetadataWriter();
-                writer.writeMetadata(evaluationSummary);
+                evaluationSummary.setJobStatus(JobStatus.SUCCESS);
             }
-            
-            CustomMetricSparkPlugin.currentlyEvaluatingContextGauge.setValue(0);
-            
-            try {
-                Boolean metricsEnabledStr = Boolean.valueOf(spark.conf().get("spark.ui.prometheus.enabled"));
-                if(metricsEnabledStr) {
-                    LOG.info("Prometheus metrics enabled, sleeping for 7 seconds to finish gathering metrics");
-                    //sleep for over 5 seconds because Prometheus only polls
-                    //every 5 seconds. If spark finishes and goes away immediately after completing,
-                    //Prometheus will never be able to poll for the final set of metrics for the spark-submit
-                    //The default promtheus config map was changed from 2 minute scrape interval to 5 seconds for spark pods
-                    Thread.sleep(7000);
-                }else {
-                    LOG.info("Prometheus metrics not enabled");
+            catch (Exception e) {
+                // If we experience an error that would make the program halt, capture the error
+                // and report it in the batch summary file
+                ByteArrayOutputStream errorDetailStream = new ByteArrayOutputStream();
+                try (PrintStream printStream = new PrintStream(errorDetailStream)) {
+                    printStream.write(e.getMessage().getBytes());
+                    printStream.write('\n');
+                    if (e.getCause() != null) {
+                        printStream.write(e.getCause().getMessage().getBytes());
+                        printStream.write('\n');
+                    }
+                    e.printStackTrace(printStream);
+                    evaluationSummary.setErrorList(Collections.singletonList(new EvaluationError(null, null, null, errorDetailStream.toString())));
+                }
+                throw e;
+            }
+            finally {
```

Comment:
I think because you have the logic to write the `EvaluationSummary` in the `finally` block, this [hunk](https://github.com/Alvearie/quality-measure-and-cohort-service/pull/228/files#diff-23b8f296c2abf0e14e68d6b1be3452c5f0717616cefdf9ce36f89f2f5d90e6afR375-R388) causes you to write the same file twice if an exception is not thrown?